### PR TITLE
We now identify players based on a ID! Less traffic. Yay.

### PR DESCRIPTION
**Pull request by BeeHiveJava**

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/MovePlayerMessage.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package nl.soccar.gameserver.message;
 
 import nl.soccar.gameserver.message.handler.MovePlayerMessageHandler;
+import nl.soccar.library.Player;
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/MovePlayerMessage.java:4]: MINOR Remove this unused import 'nl.soccar.library.Player'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/ChatMessage.java [hunk 2]
```diff
@@ -2,23 +2,30 @@
...
 public final class ChatMessage extends Message {
 
-    private final Player player;
+    private final int playerId;
+    private final Privilege privilege;
     private final String message;
 
-    public ChatMessage(Player player, String message) {
-        this.player = player;
+    public ChatMessage(int playerId, Privilege privilege, String message) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/ChatMessage.java:17]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/ChatMessage.java [hunk 3]
```diff
@@ -2,23 +2,30 @@
 
 import nl.soccar.gameserver.message.handler.ChatMessageHandler;
 import nl.soccar.library.Player;
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/ChatMessage.java:4]: MINOR Remove this unused import 'nl.soccar.library.Player'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/ChatMessage.java [hunk 4]
```diff
@@ -2,23 +2,30 @@
...
+import nl.soccar.library.enumeration.Privilege;
 import nl.soccar.socnet.message.Message;
 import nl.soccar.socnet.message.MessageConstants;
 import nl.soccar.socnet.message.MessageEvent;
 
 @MessageEvent(id = MessageConstants.CHAT_MESSAGE_ID, handler = ChatMessageHandler.class)
 public final class ChatMessage extends Message {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/ChatMessage.java:11]: MINOR Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerJoinedSessionMessage.java [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 import nl.soccar.gameserver.message.handler.PlayerJoinedSessionMessageHandler;
 import nl.soccar.library.Player;
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerJoinedSessionMessage.java:4]: MINOR Remove this unused import 'nl.soccar.library.Player'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by ghost** [comment]
SonarQube analysis reported 14 issues
* MAJOR 1 major
* MINOR 13 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ChatMessageHandler.java#L15](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/ChatMessageHandler.java#L15): The Cyclomatic Complexity of this method "handle" is 11 which is greater than 10 authorized. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MINOR [PlayerLeaveSessionMessage.java#L4](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeaveSessionMessage.java#L4): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessage.java#L4](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeftSessionMessage.java#L4): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ChatMessageHandler.java#L12](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/ChatMessageHandler.java#L12): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [MovePlayerMessageHandler.java#L4](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/MovePlayerMessageHandler.java#L4): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerJoinedSessionMessageHandler.java#L6](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerJoinedSessionMessageHandler.java#L6): Remove this unused import 'nl.soccar.library.Player'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeaveSessionMessageHandler.java#L4](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerLeaveSessionMessageHandler.java#L4): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeaveSessionMessageHandler.java#L9](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerLeaveSessionMessageHandler.java#L9): Remove this unused import 'nl.soccar.library.enumeration.TeamColour'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [PlayerLeftSessionMessageHandler.java#L4](https://github.com/PTS3-S34A/Game-Server/blob/7ec8e32d926bec8cc17e147e3ba6592e4dae18fc/Soccar [Game-Server]/src/nl/soccar/gameserver/message/handler/PlayerLeftSessionMessageHandler.java#L4): Remove this unused import 'nl.soccar.gamecommuncation.util.ByteBufUtilities'. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Review by Ranray — APPROVED** [review]

**Pull request merged**